src/Resources/views/Block/block_search_result.html.twig

```diff
@@ -59,10 +59,11 @@ file that was distributed with this source code.
                 <div class="box-body no-padding">
                     <ul class="nav nav-stacked sonata-search-result-list products-list">
                         {% for result in current_page_results %}
-                            {% set link = admin.getSearchResultLink(result) %}
-                            {% if link %}
+                            {% if admin.hasRoute('show') and admin.hasAccess('show', result) %}
```

Comment:
If I got you right, with this PR the show route will be the default route for every search result item? So if you never define any fields for show form, you can never *view* the item details?

If this is the case, I'm 👎

Reply:
> * the `fieldDescriptionFactory::create` method of `ORM` use `show` as default.
> * the `fieldDescriptionFactory::create` method of `ODM` use `edit` as default.

I know that there are some inconsistencies, but we should fix them instead of removing a big feature.

Reply:
> You can never _view_ the item details?

If you don't have fields for show form, you already don't view the item details.

And you already cannot view the item details when clicking on a link in the list of an admin since the route used is also, and already, the `show` one.

Reply:
> > * the `fieldDescriptionFactory::create` method of `ORM` use `show` as default.
> > * the `fieldDescriptionFactory::create` method of `ODM` use `edit` as default.
> 
> I know that there are some inconsistencies, but we should fix them instead of removing a big feature.
> 
> Maybe we should rename the method `getSearchResultLink` to `getDetaultItemLink` and use it everywhere

I won't call this "a big feature"
- There was multiple issue where people asked for using the `show` instead of the `edit` ; or at least configuring it.
- There is currently placed where we use `edit` if possible and `show` elsewhere ; and place where one of them is hardcoded, creating more mess and annoyance than anything else.

And I won't say the feature is removed.
In the same way you want to remove the formTheme from the admin, this logic is much better in the template than in the admin class. So people should just override the template. As I said, a `search_result` block can be added to make this easier.

Reply:
I added a `search_result` block for an easy override.